shared/src/main/scala/org/atnos/eff/Member.scala

```diff
@@ -396,41 +318,37 @@ trait MemberLower7 extends MemberLower8 {
     type Out = Fx3[T, L, M]
 
     def inject[V](r: R[V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] =
-      UnionAppendR[Fx1[T], Fx3[L, M, R], V](Union3R(r))
+      Union.appendR(Union.threeR(r))
 
-    def accept[V](union: Union[Fx3[T, L, M], V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] =
-      union match {
-        case Union3L(t) => UnionAppendL(Union1(t))
-        case Union3M(l) => UnionAppendR(Union3L(l))
-        case Union3R(m) => UnionAppendR(Union3M(m))
-      }
+    def accept[V](union: Union[Fx3[T, L, M], V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] = union.tagged match {
+      case tagged@UnionTagged(_, index) => 
+        if (index == 3) UnionAppendL(tagged.copy(index = 1))
+        else UnionAppendR(tagged.increment())
```

Comment:
Also minor, I usually provide 2 versions of these kind of methods to avoid the awkward (IMO) `()`:
```scala
def increment: UnionTagged[R, A]
def increment(i: Int) = UnionTagged[R, A]
```